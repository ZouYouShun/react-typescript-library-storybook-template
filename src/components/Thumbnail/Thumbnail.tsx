import React, { FunctionComponent, HTMLAttributes } from 'react';

type RcThumbnailProps = {
  /** is that disabled */
  disabled?: boolean;
  /** text color */
  color?: 'red' | 'blue';
  /** background color */
  background?: 'yellow' | 'green';
} & HTMLAttributes<HTMLDivElement>;

const RcThumbnail: FunctionComponent<RcThumbnailProps> = ({
  color,
  background,
  children,
  disabled,
  ...rest
}) => {
  const style = !disabled ? { color, background } : { color: 'grey' };

  return (
    <div style={style} {...rest}>
      {children}
    </div>
  );
};

RcThumbnail.defaultProps = {};

RcThumbnail.displayName = 'RcThumbnail';

export { RcThumbnail, RcThumbnailProps };
