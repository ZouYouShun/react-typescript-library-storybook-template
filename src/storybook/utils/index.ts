export * from './notShowInDocTable';
export * from './switchToControlKnobs';
