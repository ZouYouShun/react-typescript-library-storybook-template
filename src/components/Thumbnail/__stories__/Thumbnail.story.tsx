import { Meta, Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';

import { notShowInDocTable } from '../../../storybook';
import { RcThumbnail } from '../Thumbnail';

export default {
  title: 'Components/Thumbnail',
  component: RcThumbnail,
} as Meta;

type ThumbnailProps = ComponentProps<typeof RcThumbnail>;

export const Thumbnail: Story<ThumbnailProps> = ({ children, ...args }) => {
  return <RcThumbnail {...args}>{children}</RcThumbnail>;
};

Thumbnail.args = {
  children: 'Text',
  color: 'blue',
  background: 'yellow',
};

Thumbnail.argTypes = {
  ...notShowInDocTable<keyof ThumbnailProps>([]),
};
