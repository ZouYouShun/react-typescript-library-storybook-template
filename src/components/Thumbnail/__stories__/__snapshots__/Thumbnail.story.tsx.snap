// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Thumbnail Thumbnail 1`] = `
<div
  style={
    Object {
      "background": "yellow",
      "color": "blue",
    }
  }
>
  Text
</div>
`;
